models/user/email_address.go

```diff
@@ -106,12 +108,26 @@ func (email *EmailAddress) BeforeInsert() {
 	}
 }
 
+func isEmailAllowedChar(b byte) bool {
+	return (b >= '0' && b <= '9') ||
```

Comment:
It could be switched to a `switch` statement as it's more optimized for `byte`. But otherwise LGTM.